routing_model/src/scenario_tests.rs

```diff
@@ -564,6 +566,65 @@ mod dst_tests {
         );
     }
 
+    #[test]
+    fn test_parsec_expect_candidate_and_online_then_neighbour_merge() {
+        let initial_state = arrange_initial_state(
+            &initial_state_old_elders(),
+            &[
+                ParsecVote::ExpectCandidate(CANDIDATE_1).to_event(),
+                ParsecVote::Online(CANDIDATE_1).to_event(),
+            ],
+        );
+
+        run_test(
+            "Get Parsec ExpectCandidate and Online then neighbour_merge",
+            &initial_state,
+            &[Rpc::Merge.to_event()],
+            &AssertState {
+                action_our_votes: vec![ParsecVote::NeighbourMerge(NeighbourMergeInfo)],
+                ..AssertState::default()
+            },
+        );
+    }
+
+    // JON
+    #[test]
+    fn test_parsec_neighbour_merge() {
+        run_test(
+            "Get Parsec NeighbourMergeInfo",
+            &initial_state_old_elders(),
+            &[ParsecVote::NeighbourMerge(NeighbourMergeInfo).to_event()],
+            &AssertState {
+                check_and_process_elder_change_routine: CheckAndProcessElderChangeState {
+                    neighbour_merge_info: Some(NeighbourMergeInfo),
```

Comment:
Because it is a store thing, for testing, you could also enqueu it in the our_events vector in InnerAction.
You'll also actually want to store it so you can get it back out when needed.